tests/sessions_tests/tests.py

```diff
@@ -353,6 +354,32 @@ def test_session_load_does_not_create_record(self):
         # provided unknown key was cycled, not reused
         self.assertNotEqual(session.session_key, 'someunknownkey')
 
+    def test_session_save_does_not_resurrect_session_logged_out_in_other_context(self):
+        # Logged out sessions are resurrected by concurrent requests.
+
+        # This behavior is impossible to avoid with CookieSession as the session is stored in the client and
+        # there is no way to query the client about the state if it.
+        if self.backend is CookieSession:
+            raise unittest.SkipTest("CookieSessions is stored in the client and there is no way to query"
+                                    " the client about the state if it.")
+
+        # Create new session.
+        SESSION_KEY = 'tkniiqcv0js5reo8y0ofrywrhmwnt9xw'
+        s1 = self.backend(SESSION_KEY)
+        s1.save(must_create=True)
+
+        # Logout in an other context.
+        s2 = self.backend(SESSION_KEY)
+        s2.delete()
+
+        # Modify session in first context.
+        s1.modified = True
+        with self.assertRaises(UpdateError):
+            s1.save()  # This should throw an exception as the session is deleted, not resurrect it.
+
+        with self.assertRaises(Session.DoesNotExist):
+            Session.objects.get(pk=SESSION_KEY)
```

Comment:
I didn't quite get you previous comment.

That's the backend that should be queried to make sure the session has been cleared. The following should do:

```python
# make sure to set a value in s1 before calling `save(must_create=True)` on it.
s3 = self.backend(session_key)
self.assertEqual(s3.load(), {})
```